pkg/cmd/resourcesApply.go

```diff
@@ -0,0 +1,144 @@
+)
+
+// Variables set by cobra's command-line parsing.
+type ResourcesApplyCmdValues struct {
+}
+
+type ResourcesApplyCommand struct {
+	values       *ResourcesApplyCmdValues
+	cobraCommand *cobra.Command
+}
+
+// ------------------------------------------------------------------------------------------------
+// Constructors methods
+// ------------------------------------------------------------------------------------------------
+func NewResourcesApplyCommand(factory Factory, resourcesCommand GalasaCommand, rootCommand GalasaCommand) (GalasaCommand, error) {
+
+	cmd := new(ResourcesApplyCommand)
+	err := cmd.init(factory, resourcesCommand, rootCommand)
+	return cmd, err
+}
+
+// ------------------------------------------------------------------------------------------------
+// Public methods
+// ------------------------------------------------------------------------------------------------
+func (cmd *ResourcesApplyCommand) Name() string {
+	return COMMAND_NAME_RESOURCES_APPLY
+}
+
+func (cmd *ResourcesApplyCommand) CobraCommand() *cobra.Command {
+	return cmd.cobraCommand
+}
+
+func (cmd *ResourcesApplyCommand) Values() interface{} {
+	return cmd.values
+}
+
+// ------------------------------------------------------------------------------------------------
+// Private methods
+// ------------------------------------------------------------------------------------------------
+
+func (cmd *ResourcesApplyCommand) init(factory Factory, resourcesApplyCommand GalasaCommand, rootCommand GalasaCommand) error {
+
+	var err error = nil
+
+	cmd.values = &ResourcesApplyCmdValues{}
+	cmd.cobraCommand = cmd.createCobraCommand(factory, resourcesApplyCommand, rootCommand.Values().(*RootCmdValues))
+
+	return err
+}
+
+func (cmd *ResourcesApplyCommand) createCobraCommand(
+	factory Factory,
+	resourcesCommand GalasaCommand,
+	rootCommandValues *RootCmdValues,
+) *cobra.Command {
+
+	resourcesApplyCommandValues := resourcesCommand.Values().(*ResourcesCmdValues)
+	resourcesApplyCmd := &cobra.Command{
+		Use:     "apply",
+		Short:   "Apply file contents to the ecosystem.",
+		Long:    "Create or Update resources from a given file in the Galasa Ecosystem",
+		Args:    cobra.NoArgs,
+		Aliases: []string{"resources apply"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return executeResourcesApply(factory,
+				cmd, args, resourcesApplyCommandValues, rootCommandValues)
+		},
+	}
+
+	resourcesCommand.CobraCommand().AddCommand(resourcesApplyCmd)
+
+	return resourcesApplyCmd
+}
+
+func executeResourcesApply(factory Factory,
+	resourcesApplyCmd *cobra.Command,
+	args []string,
+	resourcesCmdValues *ResourcesCmdValues,
+	rootCmdValues *RootCmdValues,
+) error {
+	action := "apply"
+
+	err := loadAndPassDataIntoResourcesApi(action, factory, resourcesCmdValues, rootCmdValues)
+
+	return err
+}
+
+func loadAndPassDataIntoResourcesApi(action string, factory Factory, resourcesCmdValues *ResourcesCmdValues, rootCmdValues *RootCmdValues) error {
```

Comment:
This method doesn't load anything and pass it in. It passes the filename through. So badly named ?